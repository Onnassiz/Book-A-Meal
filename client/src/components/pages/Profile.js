import React, { Component } from 'react';
import ProfileContainer from '../container/ProfileContainer';
import NavBarContainer from '../container/NavBarContainer';

class Profile extends Component {
	render() {
		return (
			<div>
				<NavBarContainer page="caterer" />
				<ProfileContainer />
			</div>
		);
	}
}

export default Profile;
