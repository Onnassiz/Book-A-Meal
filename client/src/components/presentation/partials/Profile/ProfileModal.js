import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import empty from 'is-empty';
import { addProfileModalStyle } from './../../../../utilities/modalStyles';
import { BasicInput, TextArea } from '../../../presentation/form/BasicInput';
import ShowErrors from './../../partials/ShowErrors';
import SubmitButton from '../../../presentation/form/SubmitButton';

const closeModalStyle = {
  float: 'right',
};

const renderInputFields = data => (
  <div>
    <BasicInput
      name="businessName"
      type="text"
      label="Business Name"
      value={data.state.businessName}
      onChange={data.onChange}
      hasError={data.state.errors.businessName !== undefined}
    />
    <BasicInput
      name="mission"
      type="text"
      label="Mission Statement"
      value={data.state.mission}
      onChange={data.onChange}
      hasError={data.state.errors.mission !== undefined}
    />
    <BasicInput
      name="email"
      type="text"
      label="Business Email"
      value={data.state.email}
      onChange={data.onChange}
      hasError={data.state.errors.email !== undefined}
    />
  </div>
);

const renderModalBody = props => (
  <div>
    <div className="col-12">
      <a onClick={props.handleAddButtonClick} style={closeModalStyle}><i style={{ fontSize: 25 }} className="material-icons">close</i></a>
    </div>
    <div className="box">
      <h3>Setup Business Profile</h3>
      <ShowErrors clientErrors={props.state.errors} serverErrors={props.profile.errors || {}} />
      <form onSubmit={props.handleSubmit}>
        {renderInputFields(props)}
        <TextArea
          name="contact"
          type="text"
          label="Contact"
          value={props.state.contact}
          onChange={props.onChange}
          hasError={props.state.errors.contact !== undefined}
        />
        <SubmitButton value={empty(props.profile.businessName) ? 'Submit' : 'Update'} isLoading={props.formState.isLoading} />
      </form>
    </div>
  </div>
);

const ProfileModal = props => (
  <div>
    <Modal
      isOpen={props.state.isShowingModal}
      closeTimeoutMS={1}
      style={addProfileModalStyle}
      ariaHideApp={false}
      contentLabel="Modal"
      className="create-profile"
    >
      {renderModalBody(props)}
    </Modal>
  </div>
);

ProfileModal.propTypes = {
  state: PropTypes.object.isRequired,
};

renderModalBody.propTypes = {
  state: PropTypes.object.isRequired,
  profile: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  formState: PropTypes.func.isRequired,
  handleAddButtonClick: PropTypes.func.isRequired,
};

export default ProfileModal;
