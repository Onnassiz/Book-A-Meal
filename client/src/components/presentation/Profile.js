/* eslint-disable */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import empty from 'is-empty';
import { toast } from 'react-toastify';
import validateProfile from '../../utilities/validateProfileForm';
import ImageUploader from '../presentation/partials/ImageUploader';
import ProfileModal from './partials/Profile/ProfileModal';
import ProfileSection from './partials/Profile/ProfileSection';

class Profile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isShowingModal: false,
      isShowingDropZoneModal: false,
      mission: '',
      email: props.user.email,
      businessName: '',
      contact: '',
      errors: {},
    };
    this.handleAddButtonClick = this.handleAddButtonClick.bind(this);
    this.toggleDropZone = this.toggleDropZone.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.putImage = this.putImage.bind(this);
  }

  componentWillMount() {
    const { user, history, getProfile, profile } = this.props;
    if (user.role !== 'caterer' || !user.role) {
      history.push('/');
    }
  }

  componentDidMount() {
    const { getProfile, profile } = this.props;
    document.title = 'Profile - Just Eat';
    if (empty(profile)) {
      getProfile();
    }
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  toggleDropZone() {
    this.setState({ isShowingDropZoneModal: !this.state.isShowingDropZoneModal });
  }

  putImage(imageURL) {
    const { putImage, profile } = this.props;
    const field = {
      banner: imageURL,
    };
    putImage(profile.id, field).then((response) => {
      if (response.status === 200) {
        toast('Your profile banner has been updated');
        this.toggleDropZone(false);
      }
    });
  }

  isValid() {
    this.setState({ errors: {} });
    const { errors, isValid } = validateProfile(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  handleAddButtonClick(update) {
    if (update) {
      const { profile } = this.props;
      this.setState({
        isShowingModal: !this.state.isShowingModal,
        businessName: profile.businessName,
        email: profile.email,
        mission: profile.mission,
        contact: profile.contact,
      });
    } else {
      this.setState({ isShowingModal: !this.state.isShowingModal });
    }
  }

  createProfile(formData) {
    const { postProfile } = this.props;
    postProfile(formData).then((response) => {
      if (response.status === 201) {
        toast('Your have successfully created a profile');
        this.handleAddButtonClick(false);
      }
    });
  }

  updateProfile(formData) {
    const { updateProfile } = this.props;
    updateProfile(formData).then((response) => {
      if (response.status === 200) {
        toast('Your profile has been updated');
        this.handleAddButtonClick(false);
      }
    });
  }

  handleSubmit(e) {
    e.preventDefault();
    if (this.isValid()) {
      const { profile } = this.props;

      const formData = {
        id: profile.id,
        businessName: this.state.businessName,
        email: this.state.email,
        mission: this.state.mission,
        contact: this.state.contact,
      };

      if (empty(profile.businessName)) {
        this.createProfile(formData);
      } else {
        this.updateProfile(formData);
      }
    }
  }


  render() {
    const { formState, profile } = this.props;
    return (
      <div id="content-body">
        <div className="col-12">
          <button onClick={empty(profile.businessName) ? this.handleAddButtonClick.bind(this, false) : this.handleAddButtonClick.bind(this, true)} style={{ marginRight: 5 }} className="button">
            {empty(profile.businessName) ? 'Add Profile' : 'Update Profile'}
          </button>
          <button disabled={empty(profile.id)} onClick={this.toggleDropZone} className="button">Upload Banner</button>
        </div>

        <ProfileModal
          state={this.state}
          profile={profile}
          handleSubmit={this.handleSubmit}
          handleAddButtonClick={this.handleAddButtonClick}
          formState={formState}
          onChange={this.onChange}
        />
        <div>
          <ImageUploader putImage={this.putImage} isShowingAddPhoto={this.state.isShowingDropZoneModal} toggleAddPhoto={this.toggleDropZone} />
        </div>
        <ProfileSection profile={profile} />
      </div>
    );
  }
}

Profile.propTypes = {
  user: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  profile: PropTypes.object.isRequired,
  formState: PropTypes.object.isRequired,
  postProfile: PropTypes.func.isRequired,
  updateProfile: PropTypes.func.isRequired,
  getProfile: PropTypes.func.isRequired,
  putImage: PropTypes.func.isRequired,
};

export default Profile;
