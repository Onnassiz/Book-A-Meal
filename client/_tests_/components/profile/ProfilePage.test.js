import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Profile from '../../../src/components/pages/Profile';

const wrapper = shallow(<Profile />);

describe('<Profile />', () => {
  it('Should render Profile page successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
