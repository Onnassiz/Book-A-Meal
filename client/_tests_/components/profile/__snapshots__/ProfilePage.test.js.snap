// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> Should render Profile page successfully 1`] = `
<div>
  <Connect(NavBar)
    page="caterer"
  />
  <withRouter(Connect(Profile)) />
</div>
`;
